src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -28,62 +28,274 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Order;
 
+use Cache;
 use Cart;
+use CartRule;
+use Context;
 use Currency;
 use Order;
+use OrderCarrier;
+use OrderCartRule;
+use OrderInvoice;
+use PrestaShop\PrestaShop\Core\Cart\CartRuleData;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
 use PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision;
+use PrestaShopDatabaseException;
+use PrestaShopException;
 use Tools;
+use Validate;
 
 class OrderAmountUpdater
 {
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     */
+    public function __construct(ConfigurationInterface $configuration)
```

Comment:
Isn't a BC, the constructor is now mandatory :thinking:

Reply:
The class is new in 177 so no BC break here